articles/react-pwa/index.md

```diff
@@ -0,0 +1,465 @@
+                {todo.text}
+            </li>
+            <button onClick={HandleComplete} className="check-btn">{todo.completed?"Uncheck":"Check"}</button>
+            <button className="delete-btn" onClick={HandleDelete}>Delete</button>
+            </div>
+        </div>
+    );
+}
+
+export default Todo;
+```
+
+### Application Styling
+This tutorial assumes you know basics of CSS styling, so I have a basic CSS file: 
+
+```css
+body{
+    background-color: rgb(72, 72, 247);
+    font-size: 1.2rem;
+    max-width: 100vw;
+    max-height: 100vh;
+}
+
+input{
+    padding:1.3rem 5rem;
+    border:none;
+    font-size: 1.1rem;
+}
+.add-btn{
+    padding: 1.3rem;
+    font-size: 1.1rem;
+}
+
+button{
+    border:none;
+    margin-right: 5px;
+
+}
+
+button:hover{
+    transform: scale(0.98);
+    transition: linear 0.8s;
+    cursor: pointer;
+}
+.checked{
+    color:rgb(240, 240, 247);
+    text-decoration: line-through;
+}
+.todo{
+    display: flex;
+}
+
+.check-btn{
+    color:rgb(14, 14, 14);
+    background-color: rgb(95, 130, 245);
+}
+.delete-btn{
+    color: rgb(247, 238, 242);
+    background-color: rgb(255, 38, 38);
+}
+```
+
+
+### The React Application so far
+Your app so far should look like this:
+
+![React-PWA-image](/engineering-education/react-pwa/react-pwa1.png)
+
+![React-PWA-image2](/engineering-education/react-pwa/react-pwa2.png)
+
+
+### Adding functionality to make it a Progressive Web App
+This is the section that we will briefly configure the Progressive Web App features using the workbox tool added during create-react-app installation. [Workbox](https://developers.google.com/web/tools/workbox) is a set of tools and libraries that help to manage service workers and caching with the `CacheStorage` API.
+
+Some features of a progressive web app include:
+1. It must be discoverable and installable
+   
+2. A Progressive web App must be able to work offline
+
+3. Safety as it should rely on the HTTPS protocol 
+
+4. Should look and feel like native apps i.e run on a fullscreen mode etc.
+  
+Our application will need functionalities to run offline, caching assets by registering
+service workers, and also installing it on the end user's device home screen.
+
+### Registering a service worker
+A service worker is a script that runs in the browser's background. It handles network intercepts from requests and managing caching for offline availability. The create-react-app provided all the tools needed to create a Progressive Web App in React. To configure this to run offline, we need to register the generated service worker file. In the project, open the file named `index.js` and find the code that looks like:
+```javascript
+// If you want your app to work offline and load faster, you can change
+// unregister() to register() below. Note this comes with some pitfalls.
+// Learn more about service workers: https://cra.link/PWA
+serviceWorkerRegistration.unregister() 
+```
```

Comment:
Hi, I created a new react app using `npx create-react-app` and I'm not finding this piece of code in the `index.js` file.

Reply:
I don't find the `serviceWorker.js` file too.

Reply:
@zolomohan, I had mentioned that one should use `npx create-react-app react-pwa --template cra-template-pwa`. The --template cra-template-pwa will add it

Reply:
The files will be service-worker.js and serviceWorkerRegistration.js

Reply:
Being my first article, I apologise for all inconveniences I've caused. I will be improving on the future tutorials and writing topics that am well polished 👍.

Reply:
I didn't notice that you have updated the create command. My bad. Thanks for clearing it up.